src/convertToObject.js

```diff
@@ -5,8 +5,28 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const getDeclarationAsStrings = sourceString
+    .split(';')
+    .map((rawDeclaration) => rawDeclaration.trim());
+  const getArrayOfDeclarationEntries = getDeclarationAsStrings.map(
+    (preparedDeclaration) =>
+      preparedDeclaration
+        .split(':')
+        .map((preparedDeclarationToTrim) => preparedDeclarationToTrim.trim()),
+  );
+
+  return getArrayOfDeclarationEntries.reduce((acc, declarationEntry) => {
+    if (declarationEntry.length === 1) {
+      return acc;
+    }
+
+    return {
+      ...acc,
+      [declarationEntry[0]]: declarationEntry[1],
+    };
```

Comment:
Using the spread operator inside the `reduce` callback can lead to performance issues due to object copying in each iteration. Consider using the `Object.assign` method or direct assignment to the accumulator object to improve performance.